internal/batches/executor/check_cache.go

```diff
@@ -0,0 +1,45 @@
+package executor
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/sourcegraph/src-cli/internal/batches"
+)
+
+func CheckCache(ctx context.Context, cache ExecutionCache, clearCache bool, features batches.FeatureFlags, task *Task) (specs []*batches.ChangesetSpec, found bool, err error) {
```

Comment:
This is Not Good :tm: and I want to refactor this into another entity/thing.

Reply:
What are you thinking here?

(I wrote this before I got to your musings on the fields currently in the `Service` struct, but if there's more than just that, I'm interested!)

Reply:
I'm not sure yet. I'm playing around with some ideas in another branch. The big problem is the `TaskStatus`es and how they're woven into the executor etc.